File: mvc/src/main/java/me/kzv/simpleboardmvc/web/entity/Post.java
package me.kzv.simpleboardmvc.web.entity;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Post extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_id")
    private Long id;

    private String title;
    @Lob private String content;

    // 보여주기 여부
    private Boolean isActive;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="member_id")
    private Member member;

    @OneToMany(mappedBy = "post", cascade = CascadeType.REMOVE)
    private List<Reply> replyList = new ArrayList<>();

    @OneToMany(mappedBy = "post")
    private List<PostImg> postImgList = new ArrayList<>();

    @Builder
    public Post(String title, String content, Member member) {
        this.title = title;
        this.content = content;
        this.member = member;
        this.isActive = true;
    }

    /**
     * OneToMany 나 ManyToMany 는 디폴트가 Lazy 지연 로딩이다.
     * 하지만 ManyToOne 이나 OneToOne 은 디폴트가 eager 즉시 로딩이다.
     */
}

File: rest/backend/src/main/kotlin/me/kzv/board/web/domain/Post.kt
package me.kzv.board.web.domain

import javax.persistence.*

@Entity
class Post(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var title: String,
    @Lob
    var content: String,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    var member: Member,
    var isVisible: Boolean = true, // 관리자만 감추기 기능 추가
    // 이미지
) : BaseTimeEntity() {
    var likeCnt = 0
    var disLikeCnt = 0

    fun increaseLikeCnt() {
        likeCnt++
    }

    fun decreaseLikeCnt() {
        likeCnt--
    }

    fun increaseDisLikeCnt() {
        disLikeCnt++
    }

    fun decreaseDisLikeCnt() {
        disLikeCnt--
    }
}